iothub/device/src/InternalClient.cs

```diff
@@ -109,6 +109,7 @@ internal class InternalClient : IDisposable
         private readonly SemaphoreSlim _methodsDictionarySemaphore = new SemaphoreSlim(1, 1);
         private readonly SemaphoreSlim _deviceReceiveMessageSemaphore = new SemaphoreSlim(1, 1);
         private readonly SemaphoreSlim _moduleReceiveMessageSemaphore = new SemaphoreSlim(1, 1);
+        private readonly SemaphoreSlim _twinSemaphore = new SemaphoreSlim(1, 1);
```

Comment:
nit: rename to twinDesiredPropertySemaphore? To call out that it is relevant only for the desired property flow, and not for reported property/ get twin flow.

Reply:
sure I will rename it